include/punycode.h

```diff
@@ -0,0 +1,70 @@
+#ifndef PUNYCODE_CPP_H
+#define PUNYCODE_CPP_H
+
+#include <stdexcept>
+#include <string>
+#include <vector>
+#include <unordered_map>
+#include <unordered_set>
+
+namespace Url
+{
+
+    struct Punycoder
```

Comment:
This appears to be a namespace rather than a `struct`. All members are static. I would just change this to be a namespace if that's your intention and drop the `struct`.